zcash_client_backend/src/data_api/testing.rs

```diff
@@ -1475,7 +1477,7 @@ impl Default for TestBuilder<(), ()> {
 
 impl<A> TestBuilder<(), A> {
     /// Adds a block cache to the test environment.
-    pub fn with_block_cache<C: TestCache>(self, cache: C) -> TestBuilder<C, A> {
+    pub fn with_block_cache<C>(self, cache: C) -> TestBuilder<C, A> {
```

Comment:
This may seem controversial since it appears to be reducing the safety of the resulting `TestBuilder` (in that with this change we can't guarantee the `C` impls `TestCache` on the result), but `with_block_cache` doesn't use any of those methods internally, and `build` _also_ doesn't have this constraint. On top of that I had a genuine need to relax this.